common/src/main/java/net/adoptopenjdk/icedteaweb/StringUtils.java

```diff
@@ -67,4 +69,36 @@ public static boolean isBlank(final String str) {
         lines.add(tmp);
         return lines;
     }
+
+    /**
+     * Checks whether the first part of the given prefixString is a prefix for any of the strings
+     * in the specified array.
+     * If the {@code prefixString} contains multiple words separated by a space character, the
+     * first word is taken as prefix for comparison.
+     *
+     * @param prefixString the prefixString string
+     * @param available the strings to test
+     * @return true if the first part of the given prefixString is a prefix for any of the strings
+     * in the specified array or the specified array is empty or null, false otherwise
+     */
+    public static boolean hasPrefixMatch(final String prefixString, final String[] available) {
+        Assert.requireNonBlank(prefixString, "prefixString");
+
+        if (available == null || available.length == 0){
+            return true;
+        }
+
+        for (final String candidate : available) {
+            final String trimmedPrefix = prefixString.split(WHITESPACE_CHARACTER_SEQUENCE)[0];
```

Comment:
this can be moved before the loop to save some work.

Reply:
good catch, done